airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/MssqlDebeziumConverter.java

```diff
@@ -114,6 +117,26 @@ private void registerDate(final RelationalColumn field,
     });
   }
 
+  private void registerDateTimeOffSet(final RelationalColumn field,
+      final ConverterRegistration<SchemaBuilder> registration) {
+    registration.register(SchemaBuilder.string(), input -> {
+      if (Objects.isNull(input)) {
+        return DebeziumConverterUtils.convertDefaultValue(field);
+      }
+
+      if (input instanceof DateTimeOffset) {
+        System.out.println("running converter logic");
+        return DataTypeUtils.toISO8601String(
+            OffsetDateTime.parse(input.toString(),
+                DataTypeUtils.TIMESTAMPTZ_FORMATTER));
```

Comment:
I think you may have a problem here.
With the formatter defined as:
```kotlin
TIMESTAMPTZ_FORMATTER: DateTimeFormatter =
        DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSSSSSXXX")
```
Parsing will fail I think for any `datetimeoffset` that doesn't match this pattern - exactly six digits after the decimal point with timezone offset ('-08:30') with no space.

Reply:
Now I converted it back to localdatetime and format from that object - it would work in my test!